connectors/sync_job_runner.py

```diff
@@ -374,11 +376,15 @@ async def _sync_done(self, sync_status, sync_error=None):
 
         self.sync_job.log_info(
             f"Sync ended with status {sync_status.value} -- "
-            f"created: {result.get('doc_created', 0)} | "
-            f"updated: {result.get('doc_updated', 0)} | "
-            f"deleted: {result.get('doc_deleted', 0)} "
+            f"created: {ingestion_stats.get('doc_created', 0)} | "
+            f"updated: {ingestion_stats.get('doc_updated', 0)} | "
+            f"deleted: {ingestion_stats.get('doc_deleted', 0)} "
             f"(took {int(time.time() - self._start_time)} seconds)"  # pyright: ignore
         )
+        self.sync_job.log_info("--- Counters ---")
+        for k, v in sorted(ingestion_stats.items()):
+            self.sync_job.log_info(f"'{k}' : {v}")
+        self.sync_job.log_info("----------------")
```

Comment:
This contains duplicated counters as we already logged `created`, `updated` and `deleted`.

Considering remove these from the above log to only contain the status and time taken?

Reply:
I think that kind of exclusion is how we ended up in our current state where not all counters are logged. I think it's probably better for us to over-expose than to under-expose. Nothing is hurt by having 3 counters logged twice.

As mentioned in the threads with Artem, this is distinct from the fact that we have several counters that effectively count the same things, and we should stop doing that. That can be part of follow-up work to remove duplicate metrics, or rename metrics to more clearly highlight the differentiators. This change is just helping to expose that separate issue.

Reply:
Eventually, I expect all counters to be logged together as shown [here](https://github.com/elastic/connectors/pull/2323#issue-2218560564).

With proper renaming, the counters can be logged more clearly. Then we can simply the end-sync log to something like

```
Sync ended with status completed (took 200 seconds)
```

But it's fine if you address it as a follow-up work.

Reply:
Ideally actually we would send counters to some sort of metrics index so that the customer or us could make dashboards, run queries on them and so on, probably worth to do later?

Reply:
Yep, persisting more counters is definitely a "later" thing, since it involves changes to the protocol and to our APIs.